File: src/main/java/org/tindalos/principle/domain/detector/sdp/SDPResult.java
package org.tindalos.principle.domain.detector.sdp;

import java.util.List;

import org.tindalos.principle.domain.coredetector.CheckResult;
import org.tindalos.principle.domain.expectations.SDP;

import com.google.common.collect.Lists;

public class SDPResult implements CheckResult {

	private final List<SDPViolation> violations;
	private final SDP expectation;
	
	public SDPResult(List<SDPViolation> violations, SDP expectation) {
		this.expectation = expectation;
		this.violations = Lists.newArrayList(violations);
	}

	public List<SDPViolation> getViolations() {
		return Lists.newArrayList(violations);
	}

	public boolean expectationsFailed() {
		return violations.size() > expectation.getViolationsThreshold();
	}
	

    public Integer getThreshold() {
        return expectation.getViolationsThreshold();
    }

}

File: src/main/java/org/tindalos/principle/domain/detector/sdp/SDPResult.scala
package org.tindalos.principle.domain.detector.sdp

import org.tindalos.principle.domain.coredetector.AnalysisResult
import org.tindalos.principle.domain.expectations.SDP

class SDPResult(val violations:List[SDPViolation], private val expectation:SDP) extends AnalysisResult {
  
  val threshold = expectation.violationsThreshold
  override def expectationsFailed() = violations.length > threshold
  
  

}